File: LeetCode/LeetCode Practice Problems/1207- Unique Number of Occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        map<int,int> m;
        for(auto i : arr)
            m[i]++;
        
        set<int> s;
        for(auto i : m)
        {
            if(s.find(i.second) != s.end()) return false;
            s.insert(i.second);
        }
        
        return true;
    }
};

File: LeetCode/LeetCode Practice Problems/1207-Unique Number of Occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        int cnt[] = new int[2001];
        int freq[] = new int[1001];
        
        for(int num: arr) ++cnt[num + 1000];
        
        for(int i=0;i<2001;i++) {
            if(cnt[i] > 0)
                freq[cnt[i]]++;
            if(freq[cnt[i]] > 1) return false;
        }
        return true;
    }
}
